pkg/detectors/privatekey/privatekey.go

```diff
@@ -53,28 +55,76 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) ([]dete
 			Redacted:     token[0:64],
 		}
 
-		fingerprint, err := FingerprintPEMKey([]byte(token))
+		secret.ExtraData = make(map[string]string)
+
+		var passphrase string
+		parsedKey, err := ssh.ParseRawPrivateKey([]byte(token))
+		if err != nil && strings.Contains(err.Error(), "private key is passphrase protected") {
+			secret.ExtraData["encrypted"] = "true"
+			parsedKey, passphrase, err = crack([]byte(token))
+			if err != nil {
+				secret.VerificationError = err
+				continue
+			}
+			if passphrase != "" {
+				secret.ExtraData["cracked_encryption_passphrase"] = "true"
+			}
+		} else if err != nil {
+			// couldn't parse key, probably invalid
+			continue
+		}
+
+		fingerprint, err := FingerprintPEMKey(parsedKey)
 		if err != nil {
 			continue
 		}
 
 		if verify {
 			data, err := lookupFingerprint(fingerprint, s.IncludeExpired)
 			if err == nil {
-				secret.StructuredData = data
 				if data != nil {
 					secret.Verified = true
+					secret.ExtraData["certificate_urls"] = strings.Join(data.CertificateURLs, ", ")
 				}
+			} else {
+				secret.VerificationError = err
```

Comment:
This field isn't just informational - setting it indicates that `Verified` isn't trustworthy. We haven't yet sat down and figured out what that means when `Verified` is also set to true, which you've done later in the function. We've just been proscribing the combination entirely. (Put another way: We've added tri-state verification, but this combination is a _fourth_ state.)

The fact that this error is compatible with verifying the candidate key suggests to me that it's not actually a verification error, but possibly just another thing to put into `ExtraData`, but I could be misunderstanding what this is doing, in which case we should figure out how to update our conceptual model of indeterminacy.

(This isn't the only place in this function where you set this field in a code path that could also end up setting `Verified` to `true`, so pretend I left this comment in those other places too.)

Reply:
Let me know what you think of the change. I've collected the errors into a slice and join them if they exists and verified is false.